### ISSUE TITLE
Synthesis fan VPD

### ISSUE THREAD

**Issue by derekhoward55** [ibm-openbmc/dev#1592]
Rainier fans won't have vpd eeproms, so need to create and store in dbus.  Get pertinent info from config file of some sort.

Need to ensure vpd from hot-plugged fan is re-stored.

Ensure all necessary keywords (eg CC, PN, RI, FL) are stored.

**Comment by derekhoward55** [comment]
Thanks, yeah definitely want to make it as common to other frus as possible, just a couple complications, strawman might still be fine though (could still have the vpd in flash)-

1. need to add loc code to FL keyword in vpd, and a RID to RI keyword (phyp/OS requirement).  could have separate vpd with those filled in for each possible fan.

2. fans can be hot-swapped, so ibm-vpd-parser(?) needs to detect or be told when that happens.

Future systems will have fans with vpd that still need the FL/RI filled in, so should have flexible design to eventually allow that.

**Comment by bradbishop** [comment]
another con to the strawman is that it doesn't let us support multiple system types with a single flash image.

**Comment by derekhoward55** [comment]
yeah, "previous systems" we'd get a boilerplate from system mrw/sm with pn/ccin filled in, and dynamically fill in the ri/fl keywords, then send up chain.

**Comment by amboar** [comment]
> another con to the strawman is that it doesn't let us support multiple system types with a single flash image.

Only if you put them at a path that doesn't depend on the system type? If the system type is a component of the path then it shouldn't be a problem?

**Comment by bradbishop** [comment]
> Only if you put them at a path that doesn't depend on the system type? If the system type is a component of the path then it shouldn't be a problem?
Good point.

**Comment by spinler** [comment]
The assigned fan CCINs are:

rainier fan (2U): 7B5F
rainier blower (4U): 7B5G

FNs are still TBD.

**Comment by spinler** [comment]
The RI keyword is a 4 byte value that looks like 0xAAAABBBB, where
AAAA = resource class.  For fans it is, 0x0005
BBBB = RID, which will be 0x210x where x is the instance number of the fan

For example, for fan 2 it would be 0x00052102.

**Comment by spinler** [comment]
The plan now is to just put these 3 keywords into the VPD JSON for the fans and load them from there.  The fans already have to be in the JSON anyway so they can get location codes.   On Everest, only the RI keyword will need to be in the JSON since the fans have EEPROMs.

**Comment by msbarth** [comment]
Following added by PR https://github.ibm.com/openbmc/openbmc/pull/1775

Rainier-2U(*Part number changed for new fan hardware - tach line swap)

- PN = 02YK200
- SN = FA121A06-K28
- DR = "Fan"
- Manufacturer = "Delta"

Rainier-4U & Rainier-1S4U

- PN = 02YK323
- SN = BFM1012EH-00EYW
- DR = "Blower"
- Manufacturer = "Delta"

**Comment by derekhoward55** [comment]
https://github.ibm.com/openbmc/openbmc/pull/1790 (adding RT keywords needed by phyp) merged.

Will close this out.
